Project/Source/Components/ComponentTrail.cpp

```diff
@@ -0,0 +1,214 @@
+		if (trianglesCreated >= (maxVertices)) {
+			UpdateVerticesPosition();
+			trianglesCreated -= 30;
+		}
+
+		InsertVertex(previousPositionDown);
+		InsertTextureCoords();
+		InsertVertex(currentPositionDown);
+		InsertTextureCoords();
+		InsertVertex(previousPositionUp);
+		InsertTextureCoords();
+
+		InsertVertex(currentPositionDown);
+		InsertTextureCoords();
+		InsertVertex(currentPositionUp);
+		InsertTextureCoords();
+		InsertVertex(previousPositionUp);
+		InsertTextureCoords();
+
+		quadsCreated++;
+		Draw();
+	} else {
+		isStarted = true;
+		currentPosition = transform->GetGlobalPosition();
+		previousVectorUp = transform->GetGlobalRotation() * float3::unitY;
+		previousVectorUp.Normalize();
+		currentPositionUp = previousVectorUp * width + currentPosition;
+		currentPositionDown = -previousVectorUp * width + currentPosition;
+	}
+}
+
+void ComponentTrail::Init() {
+}
+void ComponentTrail::DrawGizmos() {
+}
+
+void ComponentTrail::OnEditorUpdate() {
+	ImGui::ResourceSlot<ResourceShader>("shader", &shaderID);
+	ImGui::InputFloat("Witdh: ", &width);
+
+	UID oldID = textureID;
+	ImGui::ResourceSlot<ResourceTexture>("texture", &textureID);
+	ResourceTexture* textureResource = App->resources->GetResource<ResourceTexture>(textureID);
+	if (textureResource != nullptr) {
+		int width;
+		int height;
+		glGetTextureLevelParameteriv(textureResource->glTexture, 0, GL_TEXTURE_WIDTH, &width);
+		glGetTextureLevelParameteriv(textureResource->glTexture, 0, GL_TEXTURE_HEIGHT, &height);
+
+		if (oldID != textureID) {
+			ComponentTransform2D* transform2D = GetOwner().GetComponent<ComponentTransform2D>();
+			if (transform2D != nullptr) {
+				transform2D->SetSize(float2((float) width, (float) height));
+			}
+		}
+		ImGui::Text("");
+		ImGui::Separator();
+		ImGui::TextColored(App->editor->titleColor, "Texture Preview");
+		ImGui::TextWrapped("Size:");
+		ImGui::SameLine();
+		ImGui::TextWrapped("%d x %d", width, height);
+		ImGui::Image((void*) textureResource->glTexture, ImVec2(200, 200));
+		ImGui::Separator();
+	}
+}
+
+void ComponentTrail::Load(JsonValue jComponent) {
+	shaderID = jComponent[JSON_TAG_TEXTURE_SHADERID];
+
+	if (shaderID != 0) {
+		App->resources->IncreaseReferenceCount(shaderID);
+	}
+
+	textureID = jComponent[JSON_TAG_TEXTURE_TEXTUREID];
+
+	if (textureID != 0) {
+		App->resources->IncreaseReferenceCount(textureID);
+	}
+}
+
+void ComponentTrail::Save(JsonValue jComponent) const {
+	jComponent[JSON_TAG_TEXTURE_SHADERID] = shaderID;
+	jComponent[JSON_TAG_TEXTURE_TEXTUREID] = textureID;
+}
+
+void ComponentTrail::Draw() {
+	unsigned int program = 0;
+	ResourceShader* shaderResouce = App->resources->GetResource<ResourceShader>(shaderID);
+	if (shaderResouce) {
+		program = shaderResouce->GetShaderProgram();
+	} else {
+		return;
+	}
+
+	glDepthMask(GL_FALSE);
+	glEnable(GL_BLEND);
+	glBlendEquation(GL_FUNC_ADD);
+	glBlendFunc(GL_ONE, GL_ONE);
+	glDisable(GL_CULL_FACE);
+
+	unsigned int quadVBO;
+	glGenBuffers(1, &quadVBO);
```

Comment:
No need to generate the VBO each frame. Cause the Vertex attributes are always in the same location. For now, you can set the quadVBO as a static variable to check if it's different of 0 to avoid the glGenBuffers each frame.

Reply:
This quad is generated each frame, that is why is there